runtime/appruntime/metadata/metadata.go

```diff
@@ -0,0 +1,66 @@
+package metadata
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"time"
+
+	gcemetadata "cloud.google.com/go/compute/metadata"
```

Comment:
This is dependent on the GCP libraries. We've introduced specific build tags for each cloud provider to ensure we don't include dependencies on cloud providers that aren't used by the application.

I think this needs to be broken out into separate packages with their own build tags, or alternatively they could be the same package but each cloud provider has its own file, and we make sure the functions only get referenced from packages that are imported only for that cloud provider.